src/pages/Sign/SignUp/SignUp.js

```diff
@@ -0,0 +1,229 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import InputBox from './InputBox';
+import CheckList from './CheckList';
+
+const SignUp = () => {
+  const [inputs, setInputs] = useState({
+    name: '',
+    phoneNumber: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+  });
+
+  const [nameValidation, setNameValidation] = useState(false);
+  const [phoneNumberValidation, setPhoneNumberValidation] = useState(false);
+  const [emailValidation, setEmailValidation] = useState(false);
+  const [passwordValidation, setPasswordValidation] = useState(false);
+  const [confirmPasswordValidation, setConfirmPasswordValidation] =
+    useState(false);
+
+  useEffect(() => {
+    if (!inputs.name.length) return;
+    checkNameValidation(inputs.name);
+  }, [inputs.name]);
+
+  useEffect(() => {
+    if (!inputs.phoneNumber.length) return;
+    inputs.phoneNumber.length <= 10
+      ? setPhoneNumberValidation(true)
+      : setPhoneNumberValidation(false);
+  }, [inputs.phoneNumber]);
+
+  useEffect(() => {
+    if (!inputs.email.length) return;
+    checkEmailValidation(inputs.email);
+  }, [inputs.email]);
+
+  useEffect(() => {
+    if (!inputs.password.length) return;
+    inputs.password.length <= 3
+      ? setPasswordValidation(true)
+      : setPasswordValidation(false);
```

Comment:
이런것도 아래와 같이 작성하면 좀 더 직관적일 수 있어요! 
```suggestion
    const condition = inputs.password.length <= 3 ;
    setPasswordValidation(condition);
```